### ISSUE TITLE
Error handling for missing loader or missing chart package

### ISSUE THREAD

**Issue by naoak** [GoogleWebComponents/google-chart#123]
Hi. 
I'd like to handle the title error cases.

An extreme case, Great Firewall blocks any connections to gstatic.com. Is there any way to handle this kind of errors?

**Comment by naoak** [comment]
Additionally,  if gstatic.com site is down,  loading loader.js blocks whole scripting until it is timeouted. I think we should load external scripts by IronJsonpLibraryBehavior as it used to be.

**Comment by naoak** [comment]
I tried rewriting the codes by using IronJsonpLibrary, but it did not go well because  `loader.js`  does not support jsonp loading...

**Comment by wesalvaro** [comment]
Yes, that was the original reason that the google-api-loader was dumped. But then the `loader.js` did not support being loaded more than once...

We could do some magic, but not being able to access `gstatic.com` is probably an issue specific to _only_ the great firewall... so it doesn't really make sense to me to do so in this component. I think it makes the most sense for the new loader to have a compatible component that we could make use of here.

Until then, you may just need to fork your own copy that is able to work around it until a permanent solution is available.

**Comment by naoak** [comment]
@wesalvaro Thanks for your clarification and suggestion. Actually, I've already created its fork [here](https://github.com/naoak/google-chart/commit/d94a9744b7fdd14413fd22853935b91994aa4dd9), that works fine for now. I've not tried, but I might be able to do the similar by lazy loading google-chart component itself.

**Comment by naoak** [comment]
Unfortunately, I've decided to not use google chart due to the performance issue. I was using this component for drawing a simple line chart, but this component loaded so many library files which are totally almost 1MB without gzipped. I could not ignore the payload just for the one chart.

I might not participate, but I hope google-chart for further development. If you think you don't need this feature, please feel free to close. Thank you till now.

**Comment by wesalvaro** [comment]
I'm not sure how that's possible. It loads only the necessary amount of data for Google Visualizations for a line chart.

Here is my network panel for the chart demo page. The demo page loads all possible chart packages and data for demo purposes.
![Loading all packages](https://cloud.githubusercontent.com/assets/13286/17428077/8eb63fea-5b20-11e6-88d7-aabcf05e1268.png)
It only loads ~425KB for everything. Most of it is in the UI module.
Here is the network panel for loading only a line chart.
![Loading only Line chart](https://cloud.githubusercontent.com/assets/13286/17428206/c439baec-5b21-11e6-86c8-b36857d51462.png)
It only loads ~325KB for a line chart.
The content looks gzip encoded to me, too.

Of course, maybe the biggest part of these downloads is features you are not directly utilizing, features like DataTables, DataViews, Queries, Formatters, localization, cross-browser support etc.

For a simple line chart there are probably other much smaller and still sufficient libraries, though.
Which did you decide to go with?
